datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -1159,11 +1157,11 @@ fn replace_order_preserving_variants_helper(
     if let Some(repartition) = exec_tree.plan.as_any().downcast_ref::<RepartitionExec>() {
         if repartition.preserve_order() {
             return Ok(Arc::new(
+                // new RepartitionExec don't preserve order
                 RepartitionExec::try_new(
                     updated_children.swap_remove(0),
                     repartition.partitioning().clone(),
-                )?
-                .with_preserve_order(false),
```

Comment:
Per @ozankabak 's suggestion, I removed the `with_preserve_order(false)` as well as the boolean argument which was confusing. Now `with_preserve_order()` is only called when the code is attempting to maintain the order